packages/cli/lib/sync.unit.test.ts

```diff
@@ -451,4 +469,40 @@ describe('generate function tests', () => {
         expect(modelsFile).toContain(`reference: Other[];`);
         expect(modelsFile).toContain(`nullableDate: Date | null;`);
     });
+
+    it('should be able to compile files in nested directories', async () => {
+        await fs.promises.rm(testDirectory, { recursive: true, force: true });
+        await fs.promises.mkdir(testDirectory, { recursive: true });
+        await copyDirectoryAndContents(`${fixturesPath}/nango-yaml/v2/nested-integrations/hubspot`, './hubspot');
+        await copyDirectoryAndContents(`${fixturesPath}/nango-yaml/v2/nested-integrations/github`, './github');
+        await fs.promises.copyFile(`${fixturesPath}/nango-yaml/v2/nested-integrations/nango.yaml`, `./nango.yaml`);
+
+        const success = await compileService.run({ debug: true });
+
+        await fs.promises.rm('./hubspot', { recursive: true, force: true });
+        await fs.promises.rm('./github', { recursive: true, force: true });
+        await fs.promises.rm('./dist', { recursive: true, force: true });
+        await fs.promises.rm('./nango.yaml', { force: true });
+        await fs.promises.rm('./models.ts', { force: true });
+
+        expect(success).toBe(true);
```

Comment:
there is no verification that nested file have been indeed compiled no?

Reply:
No, but if not compiled successfully, success won't be true.